tests/sources/generic.py

```diff
@@ -21,33 +21,50 @@
 async def main(queue: asyncio.Queue, args: Dict[str, Any]):
     payload = args.get("payload")
     randomize = args.get("randomize", False)
+    display = args.get("display", False)
     add_timestamp = args.get("timestamp", False)
     create_index = args.get("create_index", None)
-    delay = int(args.get("delay", 0))
-    loop_count = int(args.get("loop_count", 1))  # -1 infinite
+
+    startup_delay = int(args.get("startup_delay", 0))
+    event_delay = int(args.get("event_delay", 0))
+    repeat_delay = int(args.get("repeat_delay", 0))
     loop_delay = int(args.get("loop_delay", 0))
+    shutdown_after = int(args.get("shutdown_after", 0))
+
+    loop_count = int(args.get("loop_count", 1))  # -1 infinite
+    repeat_count = int(args.get("repeat_count", 1))
 
     if not isinstance(payload, list):
         payload = [payload]
 
     iteration = 0
     index = 0
+
+    await asyncio.sleep(startup_delay)
+
     while iteration != loop_count:
         if loop_delay > 0 and iteration > 0:
             await asyncio.sleep(loop_delay)
         if randomize:
             random.shuffle(payload)
         for event in payload:
-            extras = {}
-            if create_index:
-                extras[f"{create_index}"] = index
-            if add_timestamp:
-                extras["timestamp"] = str(datetime.now())
-            event.update(extras)
-            await queue.put(event)
-            await asyncio.sleep(delay)
-            index += 1
+            for _ in range(repeat_count):
+                data = {}
+                if create_index:
+                    data[f"{create_index}"] = index
```

Comment:
This is an odd syntax for a dict key, why not force the string type of `create_index` in line 26 and set the dict key in the regular way?